### ISSUE TITLE
Code Request: rMD17 Student Training Script and Loss Settings to Reproduce Paper Results

### ISSUE THREAD

**Issue by KunhuanLiu** [HongxinXiang/EDG#2]
Hi team,

I’m trying to reproduce the rMD17 results from the paper. Right now, the provided finetune script is tailored to QM9: `EDG/EDG/finetune_QM9_EDG.py` asserts `args.dataset == "QM9"` and uses QM9-specific splits, so it can’t be run on rMD17 out of the box. I do see you’ve provided rMD17 dataset loaders, including a distillation-ready variant (`Geom3D/datasets/dataset_rMD17_distillation.py` and `Geom3D/datasets/dataset_rMD17.py`, exported via `DatasetrMD17`), which makes it look like rMD17 student training is intended to be supported.

Could you clarify how you trained the student models on rMD17 and, if possible, share the corresponding code or command lines? I have a few implementation questions particular:
- Loss function: For rMD17, did you train with both energy and forces? If so, what prefactors did you use for energy vs. force for each 3D model? I see defaults in `EDG/EDG/config_distillation.py` (`--md17_energy_coeff=0.05`, `--md17_force_coeff=0.95`). Were these the exact values used for rMD17 in the paper, and were they uniform across models?
- Normalization: Did you enable `--energy_force_with_normalization` during training on rMD17?
- ED guidance: I saw the `--weight_ED` for the ED-guided auxiliary loss for the student training. Did you just use the default `--weight_ED` value (=1)?
- Splits: Can you confirm that you used the default `--rMD17_split_id` (=01) in the paper runs?

Lastly, it would be tremendously helpful if you would be able to provide the student training code (or an example script) analogous to `finetune_QM9_EDG.py`, but for rMD17 (using `DatasetrMD17`, rMD17 splits, energy+force losses, and optional ED details). 

Having the official rMD17 finetune code or a reference command would ensure faithful reproduction of the paper’s numbers. Thanks a lot for your help and for releasing the dataset loaders and teacher components.

Best,

**Comment by HongxinXiang** [comment]
Thanks for your interest! I'll be releasing the rMD17 code in the next few days, so you can check out all the details. I'll let you know once I release it.

**Comment by HongxinXiang** [comment]
The code has been uploaded to: https://github.com/HongxinXiang/EDG/blob/master/EDG/finetune_rMD17_EDG.py

You can also find instructions on running finetune_rMD17_EDG.py in the README.

I only checked the correctness of the code logic, but I didn't run it to verify it. Please let me know if you encounter any problems during the run.

**Comment by KunhuanLiu** [comment]
Thank you very much for sharing your code for finetuning on rMD17 dataset.

I have a couple of questions:

1. I see you used the `energy_force_with_normalization` option, which modifies the predicted energy as follows:
    
    ```
    if args.energy_force_with_normalization:
        pred_energy = pred_energy * FORCE_MEAN_TOTAL + ENERGY_MEAN_TOTAL * NUM_ATOM
    ```
    
    Based on my understanding, this translates to:
    
    ```
    predicted energy of a molecule = raw model energy output × average RMS(atomic force) + average atomic energy × number of atoms of the molecule
    ```
    
    Here, `average RMS(atomic force)` and `average atomic energy` are averages across the training dataset. The second term seems to represent an "expected total energy" based on the molecule's atom count, which makes sense. However, I'm confused by the first term (scaling the raw energy output by the `FORCE_MEAN_TOTAL`).
    
    Would you mind explaining the rationale for this part of the normalization?
    
2. I also wanted to reproduce the VisNet results from the paper, but I didn't find the VisNet model implementation in this codebase. Did you use the implementation from the `torch_geometric` package or another source? I would appreciate any details you can provide to help me reproduce those results.

**Comment by HongxinXiang** [comment]
Sorry for the delayed response.

Regarding question 1, please refer to Appendix J.2 of the paper (https://arxiv.org/pdf/2306.09375), where you should be able to find the answer.

As for question 2, we have made modifications in the https://github.com/microsoft/AI2BMD/tree/ViSNet branch. However, since the cloud server we used is currently inaccessible, we will release the corresponding code once it becomes available. In the meantime, you can refer to the current rMD17 implementation and adapt it accordingly. Most of the work involves transferring the distillation-related code from rMD17 to the ViSNet project, which should require only minimal adjustments.

**Comment by HongxinXiang** [comment]
The EDG version of VisNet has been released, see EDG-for-VisNet directory.
